### PR TITLE
rustdoc: Give a more accurate span for anchor failures

### PR DESCRIPTION
**Pull request by jyn514**
r? @euclio

### CODE REVIEW

File: src/librustdoc/passes/collect_intra_doc_links.rs [hunk 1]
```diff
@@ -1957,19 +1957,27 @@ fn resolution_failure(
...
 
     report_diagnostic(cx.tcx, BROKEN_INTRA_DOC_LINKS, &msg, &diag_info, |diag, sp| {
-        if let Some(sp) = sp {
+        if let Some(mut sp) = sp {
+            if let Some((fragment_offset, _)) =
+                diag_info.ori_link.char_indices().filter(|(_, x)| *x == '#').nth(anchor_idx)
+            {
+                sp = sp.with_lo(sp.lo() + rustc_span::BytePos(fragment_offset as _));
+            }
             diag.span_label(sp, "contains invalid anchor");
```
Comment (euclio) [inline src/librustdoc/passes/collect_intra_doc_links.rs:1981]: Looks like we should change this to just "invalid anchor" now.

Comment (jyn514) [reply src/librustdoc/passes/collect_intra_doc_links.rs]: Thanks, done.

**Comment by euclio** [comment]
I don't have bors privileges, but LGTM.

**Comment by jyn514** [comment]
Thanks for the review!

@bors r=euclio

**Pull request merged**